src/java.base/unix/classes/sun/nio/fs/UnixCopyFile.java

```diff
@@ -217,6 +220,24 @@ private static void copyDirectory(UnixPath source,
         }
     }
 
+    // calculate the least common multiple of two values
+    private static long lcm(long x, long y) {
+        if (x <= 0 || y <= 0)
+            throw new IllegalArgumentException("Non-positive parameter");
+
+        long u = x;
+        long v = y;
+
+        while (u != v) {
+            if (u < v)
```

Comment:
It might be worth a comment about the expected inputs so it is known to converge quickly.
In the general case it might wrap around and go unstable.

Reply:
Comment added in f151dc4cf6f4e59fff2d2a0f486929940cafb1ae.